Use the bundler github shortcut in Gemfile

octopusinvitro opened the pull request:
# What does this do?

Uses the `github:` shortcut in the `Gemfile` for gems that are not in Rubygems and we have to link to their repos.

# Why was this needed?

For security reasons as described in http://bundler.io/git.html#security

# Relevant Issue(s)

Part of https://github.com/everypolitician/everypolitician/issues/518

# Implementation notes

While making #18495 Dave and I discovered that bumping the `everypolitician-popolo` form 0.6 to 0.7 was a breaking change. This PR follows a different approach by changing one gem at a time and checking that nothing is broken.

Steps followed:

* Adding 
```ruby
git_source(:github) { |repo_name| "https://github.com/#{repo_name}.git" }
``` 
to the top of the Gemfile and running `bundle install` without changing anything else produces the error:
```
Could not find gem 'everypolitician-popolo (~> 0.6.0)' in https://github.com/everypolitician/everypolitician-popolo.git (at master@9cf0704).
Source contains 'everypolitician-popolo' at: 0.7.0 ruby
```

The error was fixed by adding the SHA as a `ref:` to the 0.6 tag at the end of the line:
```ruby
gem 'everypolitician-popolo', '~> 0.6.0', github: 'everypolitician/everypolitician-popolo', ref: '63c7a13'
```
Running `bundle install` updates all the other gems to point to master due to the line added at the top of the Gemfile, as shown in 01e74b5.

* To test if anything was broken by this change, `bundle exec rake clean default` was ran against `Pakistan/Assembly` two times, one without changing anything, the second using a new `data.csv` file produced from the `data.sqlite` file generated from the official scraper with [this changes](https://github.com/everypolitician-scrapers/pakistan-national-assembly/pull/2):
 - First run: no errors and no files changed.
 - Second run: no errors, expected files changed.

* Change the close_old_pull_requests gem to secure git url
 - First run: no errors and no files changed.
 - Second run: no errors, expected files changed.

* Change the everypolitician-pull_request gem to secure gt url
 - First run: no errors and no files changed.
 - Second run: no errors, expected files changed.

* Bump popolo to 0.7 and remove the `ref:` at the end of the line:
```ruby
gem 'everypolitician-popolo', '~> 0.7.0', github: 'everypolitician/everypolitician-popolo'
```
   First run: Produces the error:
```bash
rake aborted!
NoMethodError: undefined method `memberships' for nil:NilClass
everypolitician-data/rake_build/generate_final_csvs.rb:17:in `block (3 levels) in <top (required)>'
everypolitician-data/rake_build/generate_final_csvs.rb:15:in `each'
everypolitician-data/rake_build/generate_final_csvs.rb:15:in `block (2 levels) in <top (required)>'
/home/admin/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/home/admin/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
Tasks: TOP => default => csvs => term_csvs:term_tables
(See full trace by running task with --trace)
```
 Second run: Same error

As a result, the last change is not included in this PR since bumping the version number in EP-Popolo seems to be a breaking change.

# Screenshots

None

# Notes to Reviewer

None

# Notes to Merger

None

tmtmtmtm commented:
@octopusinvitro https://github.com/everypolitician/everypolitician-popolo/pull/107 should have fixed the problem you were having.

PS: I think the PR needs a more descriptive name.

octopusinvitro commented:
@tmtmtmtm I have updated the title of the PR and added a commit to bump the EP-popolo gem version from 0.6 to 0.7.

I checked the build against Pakistan again, got no errors and no files changed.

tmtmtmtm commented:
@octopusinvitro this has significant merge conflicts that are proving very difficult to resolve, as it appears to be based off a very old version of `master`, which has since had other changes to `Gemfile.lock`. I've redone it from scratch as a new PR: https://github.com/everypolitician/everypolitician-data/pull/18610
